src/site/content/en/blog/google-search-sw/index.md

```diff
@@ -0,0 +1,463 @@
+service worker: in Chrome, [a parsed, byte code representation](https://v8.dev/blog/code-caching-for-devs#use-service-worker-caches)
+of that JavaScript is stored and reused, leading to less work that needs to be
+done at runtime in order to execute the JavaScript on the page.
+
+## Non-goals
+
+While many in the web development community associate service workers with [Progressive Web Apps](https://developers.google.com/web/progressive-web-apps/),
+building a "Google Search PWA" was not an initial goal of the team. The Google
+Search web app doesn't currently provide metadata via a
+[web app manifest](https://developers.google.com/web/fundamentals/web-app-manifest/),
+nor does it encourage users to go through the
+[Add to Homescreen flow](https://developers.google.com/web/fundamentals/app-install-banners/).
+The Search team is currently satisfied with users coming to their web app via
+the traditional entry points for Google Search.
+
+Rather than trying to turn the Google Search web experience into the equivalent
+of what you'd expect from an installed application, the focus on the initial
+roll out was to progressively enhance the existing web site.
+
+## Challenges and solutions
+
+Here are a few of the hurdles that needed to be overcome in order to achieve the
+team's stated goals. While some of these challenges are specific to Google
+Search, many of them are applicable to a wide range of sites that might be
+considering a service worker deployment.
+
+### Service worker overhead
+
+The biggest challenge, and the one true blocker for launching a service worker
+on Google Search, was to ensure that it did not do anything that might increase
+user-perceived latency. Google Search takes performance _very_ seriously, and in
+the past, has blocked launches of new functionality if it contributed even tens
+of milliseconds of additional latency for a given user population.
+
+When the team started collecting performance data during their earliest
+experiments, it became obvious that there would be a problem. The HTML returned
+in response to
+[navigation requests](https://developers.google.com/web/fundamentals/primers/service-workers/high-performance-loading#first_what_are_navigation_requests)
+for the search result page is dynamic, and varies greatly depending on logic
+that needs to run on Search's web servers. There's currently no way for the
+service worker to replicate this logic and return cached HTML immediately—the
+best it could do is to pass along navigation requests to the backend web
+servers, which necessitates a network request. Without a service worker, this
+network request would happen immediately upon user navigation, but when a
+service worker is registered, it will (by default) always need to be started up
+and given a chance to execute its [`fetch event handlers`](https://developers.google.com/web/fundamentals/primers/service-workers/#cache_and_return_requests),
```

Comment:
Suggestion to break up the run-on sentence a bit, and try to lean. Something like:

Without a service worker, this network request happens immediately upon user navigation. When a service worker is registered, it always needs to be started up and given a chance to execute its fetch event handlers, even when there's no chance those fetch handlers will do anything other than go to the network.